pkg/webhook/webhook.go

```diff
@@ -55,15 +59,49 @@ func (q *QuayIntegrationMutator) Handle(ctx context.Context, req admission.Reque
 			}
 		}
 	} else {
-
-		admissionResponse = getAdmissionResponseForBuild(build, &quayIntegration)
+		// Check if builder service account has secret
+		hasSecret, serviceAcctErr := q.checkSecretForBuilderServiceAccount(ctx, &req, &quayIntegration)
+
+		if !hasSecret {
+			if serviceAcctErr != nil {
+				admissionResponse = &admissionv1.AdmissionResponse{
+					Allowed: false,
+					Result: &metav1.Status{
+						Message: serviceAcctErr.Error(),
+					},
+				}
+			} else {
+				admissionResponse = &admissionv1.AdmissionResponse{
+					Allowed: false,
+				}
+			}
+		} else {
+			admissionResponse = getAdmissionResponseForBuild(build, &quayIntegration)
+		}
 
 	}
 
 	return admission.Response{AdmissionResponse: *admissionResponse}
 
 }
 
+func (q *QuayIntegrationMutator) checkSecretForBuilderServiceAccount(ctx context.Context, ar *admission.Request, quayIntegration *quayv1.QuayIntegration) (bool, error) {
+
+	builderServiceAccnt := &corev1.ServiceAccount{}
+
+	err := q.Client.Get(ctx, types.NamespacedName{Namespace: ar.Namespace, Name: string(qotypes.BuilderOpenShiftServiceAccount)}, builderServiceAccnt)
+	if err != nil {
+		logging.Log.Info("Failed to get existing builder service account")
```

Comment:
This message won't be useful on a real cluster, it doesn't include the namespace nor the error.

Reply:
modified the message to include namespace